SofaKernel/framework/sofa/core/behavior/BaseMechanicalState.h

```diff
@@ -22,17 +22,14 @@
 #ifndef SOFA_CORE_BEHAVIOR_BASEMECHANICALSTATE_H
 #define SOFA_CORE_BEHAVIOR_BASEMECHANICALSTATE_H
 
-#include <sofa/config/build_option_experimental_features.h>
```

Comment:
A very bad things is that even in condition like #if MACRO == 1
When the macro is not defined then its value is == 0 so removing the include line is silent.
https://gcc.gnu.org/onlinedocs/cpp/If.html

A workaround I can propose it to do:
```cpp
#define MYTEST() 0 
....
#if MYTEST() 
        std::cout << "COUCOU A" << std::endl ;
#else
        std::cout << "COUCOU B" << std::endl ;
#endif 
```

So that we get error when MYTEST is not defined (the bad things is that the error is not very clear with some compilers):
with gcc:
toto.cpp:8:11: error: missing binary operator before token "(" 
 #if MYTEST()

with clang:
toto.cpp:8:5: error: function-like macro 'MYTEST' is not defined
#if MYTEST() 
    ^